tsdb/head.go

```diff
@@ -813,20 +827,24 @@ func (h *Head) Truncate(mint int64) (err error) {
 	if err := h.chunkDiskMapper.Truncate(mint); err != nil {
 		return errors.Wrap(err, "truncate chunks.HeadReadWriter")
 	}
+	return nil
+}
 
-	if h.wal == nil {
+// truncateWAL removes old data before mint from the WAL.
+func (h *Head) truncateWAL(mint int64) error {
+	if h.wal == nil || mint <= h.lastWALTruncationTime.Load() {
 		return nil
 	}
-	start = time.Now()
+	start := time.Now()
+	h.lastWALTruncationTime.Store(mint)
```

Comment:
Shouldn't be stored only in case of success?

Reply:
I gave it a thought but felt we should not allow multiple attempts to truncate WAL, because if it failed for some reason, it's less likely it will succeed in the next attempt (with disk filling up is the most common failure that I see). So, with this, next truncation attempt will be done after next Head compaction.

But it would not make a difference here because truncation of WAL is tried only once after Head compaction.